pthread_stop_world.c

```diff
@@ -325,10 +327,53 @@ GC_INLINE void GC_store_stack_ptr(GC_stack_context_t crtn)
 # endif
 }
 
+static int find_segment(struct dl_phdr_info *info, size_t size, void *data) {
+    UNUSED_ARG(size);
+    tlr * roots = (tlr *) data;
+
+    for (size_t i = 0; i < info->dlpi_phnum; i ++) {
```

Comment:
Out of curiosity, of those things we're iterating over, what parts can actually change at runtime? I *guess* it's `info->dlpi_tls_data` that varies? If that's the only thing, could we:

1. Cache the result of iteration and just store an array of pointers that's basically `ptr_t **` (I guess we also have to store the sizes in a sister `size_t *` array) and thus we could avoid the probably fairly slow `dl_iterate` costs?
2. Is this enough of a cost that such caching would be worthwhile?

It's fine if we don't address this in this PR, but I thought I'd ask.

Reply:
For this PR, I was more concerned about correctness. I'm happy to address these in a later PR if you think it's worthwhile.